compiler/rustc_incremental/src/persist/save.rs

```diff
@@ -13,9 +13,10 @@ use super::file_format;
 use super::fs::*;
 use super::work_product;
 
-/// Save and dump the DepGraph.
+/// Saves and writes the [`DepGraph`] to the file system.
 ///
-/// No query must be invoked after this function.
+/// This function should only run after all queries have completed.
+/// Attempting to run a query after saving the dependency would be a problem.
```

Comment:
More precisely: this function saves both the dep-graph and the query result cache, and drops the result cache. Trying to execute a query afterwards would attempt to read the result cache we just dropped.